internal/bufferedwrites/upload_handler.go

```diff
@@ -125,16 +125,16 @@ func (uh *UploadHandler) Finalize() error {
 		// small writes of size less than 1 block.
 		err := uh.createObjectWriter()
 		if err != nil {
-			return fmt.Errorf("createObjectWriter failed for object %s: %w", uh.objectName, err)
+			return nil, fmt.Errorf("createObjectWriter failed for object %s: %w", uh.objectName, err)
 		}
 	}
 
-	err := uh.writer.Close()
+	obj, err := uh.bucket.FinalizeUpload(context.Background(), uh.writer)
 	if err != nil {
 		logger.Errorf("UploadHandler.Finalize(%s): %v", uh.objectName, err)
```

Comment:
if we dont log and return wont it get logged somewhere? its not mandatory to log when returning the error.

Reply:
removed, thanks. The other failure log in this file is for uploader go routine (async) so not removing it.